sphinx/source/developers/threading.rst

```diff
@@ -0,0 +1,30 @@
+Threading
+=========
+
+Consistency
+-----------
+
+CCF guarantees session consistency from the perspective of a :term:`TLS` client connection.
+This means that within a single client connection a client is guaranteed to read its own writes.
```

Comment:
That's not true at the moment, because of write forwarding. A client can send a write, which will get forwarded to the primary, followed by a read, which won't and which will get executed locally against earlier state.

We've discussed this before, and consensus was that forwarding should be a per-session toggle, ie. once a command has been forwarded on a session, all subsequent commands must be too.

There could be additional logic in the forwarding code that monitors the commit status of the latest write and flips the toggle back once that's replicated on the node. That's not necessary for session consistency but it would be an optimisation.

Reply:
Thank you, updating the text

Reply:
@ashamis filed #881 for this.